streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -568,6 +568,10 @@ void suspend(final boolean clean,
                 commit(false);
             } finally {
                 if (eosEnabled) {
+                    // In #commit, we don't do checkpoint file writing when eos is turned on. To avoid race condition when writing checkpoint file
+                    // through StateManager #closeSuspendedwe, we decide to always checkpoint offsets after commit for EOS.
+                    // The check here is to avoid double checkpoint file write through #commit and #suspend.
```

Comment:
Which check? I don't see any `if` clause.

Reply:
I was thinking the same. Did you mean that the other change on this PR eliminates the chance for a double checkpoint file writes?

Reply:
Oh, the check is just pointing to `eosEnabled`. Let me update the comment.